File: leetcode/Algorithms/388.LongestAbsoluteFilePath/Solution.py
class Solution(object):
    def lengthLongestPath(self, input):
        """
        :type input: str
        :rtype: int
        """
        stack = [0] * (input.count('\n') + 2)
        max_length = 0
        for name in input.split('\n'):
            current_indent = name.rfind('\t') + 1;
            stack[current_indent + 1] = stack[current_indent] + len(name) - current_indent + 1
            if name.find(".") > -1:
                max_length = max(max_length, stack[current_indent + 1] - 1)

        return max_length

File: leetcode/Algorithms/388.LongestAbsoluteFilePath/Solution.java
public class Solution {
    public int lengthLongestPath(String input) {
        Stack<String> store = new Stack<>();

        int result = 0, length = 0;
        for (String line: input.split(String.valueOf('\n'))){
            int depth = line.lastIndexOf('\t') + 2;
            String str = line.substring(depth - 1);

            while (store.size() >= depth && !store.isEmpty()) {
                length -= store.pop().length() + 1;
            }

            store.push(str);
            length += str.length() + 1;

            if (line.contains(".")) result = Math.max(result, length - 1);
        }

        return result;
    }
}
